Source/ARTAuth.m

```diff
@@ -337,14 +337,25 @@ - (void)createTokenRequest:(ARTTokenParams *)tokenParams options:(ARTAuthOptions
     }
 
     if (mergedOptions.queryTime) {
-        [_rest time:^(NSDate *time, NSError *error) {
-            if (error) {
-                callback(nil, error);
-            } else {
-                mergedTokenParams.timestamp = time;
-                callback([mergedTokenParams sign:mergedOptions.key], nil);
-            }
-        }];
+        if (self.timeOffset == nil) {
+            [_rest time:^(NSDate *time, NSError *error) {
+                if (error) {
+                    callback(nil, error);
+                } else {
+                    NSDate *dateNow = [NSDate date];
+                    long long nowMillis = [dateNow timeIntervalSince1970]*1000;
```

Comment:
Why do you convert the `NSTimeInterval`s to `long long` and then to `NSNumber` instead of just storing the time offset as a `NSTimeInterval`? And why do you multiply by 1000 instead of working with seconds with decimals, which is what `NSTimeInterval` is?